### PR TITLE
(MODULES-2659) Detect CCM pending reboot state

### PR DESCRIPTION
**Pull request by Iristyle**
- Use WMI to query the CCM_ClientUtilities class of the CCM ClientSDK
  namespace to call the DetermineIfRebootPending WMI method.  The object
  returned contains a ReturnValue and two flags RebootPending and
  IsHardRebootPending

  See the Configuration Manager Client SDK WMI reference for more info:
  https://msdn.microsoft.com/en-us/library/jj885518.aspx

  Only unit tests are included since writing an acceptance test requires
  additional dependencies that are difficult to install in this context.
  To manual test, install the System Center Configuration Manager Client

### CODE REVIEW

File: lib/puppet/provider/reboot/windows.rb [hunk 1]
```diff
@@ -175,6 +176,24 @@ def pending_dsc_reboot?
...
+    require 'win32ole'
+    root = 'winmgmts:\\\\.\\root\\ccm\\ClientSDK'
+    reboot = false
+
+    begin
+      ccm = WIN32OLE.connect(root)
+
+      ccm_client_utils = ccm.Get('CCM_ClientUtilities')
+
+      pending = ccm_client_utils.ExecMethod_('DetermineIfRebootPending')
```
Comment (ferventcoder) [inline lib/puppet/provider/reboot/windows.rb]: That name tho...

**Comment by ferventcoder** [comment]
:+1: LGTM

File: lib/puppet/provider/reboot/windows.rb (hunk 1 above)
Comment (Iristyle) [inline lib/puppet/provider/reboot/windows.rb]: Variable name, or WMI method name?

Comment (ferventcoder) [inline lib/puppet/provider/reboot/windows.rb]: Method name. It makes too much sense ha :)

**Pull request merged**